File: app/src/main/java/com/movies/popular/popularmovies/data/common/mappers/Mappers.java
package com.movies.popular.popularmovies.data.common.mappers;

import android.support.annotation.Nullable;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * Created with Android Studio.
 * User: Sasha Shcherbinin
 * Date: 9/4/17
 * Time: 7:50 PM
 */

public final class Mappers {
    public Mappers() {
    }

    public static <F, T> List<T> mapCollection(@Nullable List<F> list, Mapper<F, T> mapper) {
        if(list == null) {
            return Collections.emptyList();
        } else {
            int size = list.size();
            ArrayList result = new ArrayList(size);

            for(int i = 0; i < size; ++i) {
                T map = mapper.map(list.get(i));
                if (map != null) {
                    result.add(map);
                }
            }

            return result;
        }
    }
}

File: library/src/main/java/com/sprinklebit/library/data/common/mappers/Mappers.kt
package com.sprinklebit.library.data.common.mappers

import java.util.*
import java.util.Collections.emptyList

/**
 * Created with Android Studio.
 * User: Sasha Shcherbinin
 * Date: 9/4/17
 * Time: 7:50 PM
 */
object Mappers {

    @JvmStatic
    fun <F, T> mapCollection(list: List<F>?, mapper: Mapper<F, T>): List<T> {
        return if (list == null) {
            emptyList()
        } else {
            val size = list.size
            val result = ArrayList<T>(size)

            for (i in 0 until size) {
                val map = mapper.map(list[i])
                if (map != null) {
                    result.add(map)
                }
            }

            result
        }
    }
}
